cupy/lib/polynomial.pyx

```diff
@@ -161,11 +161,13 @@ cdef class poly1d:
                             'addition is not supported')
         return _routines_poly.polyadd(self, other)
 
-    # TODO(Dahlia-Chehata): implement using polymul
     def __pow__(self, val, modulo):
         if not cupy.isscalar(val) or int(val) != val or val < 0:
             raise ValueError('Power to non-negative integers only.')
-        raise NotImplementedError
+        out = 1
+        for _ in range(val):
+            out = _routines_poly.polymul(self, out)
```

Comment:
How about trying O(log(val)) time convolve calls, and comparing performances?

Reply:
note: https://en.wikipedia.org/wiki/Exponentiation_by_squaring

Reply:
I'll make performance comparison some time later.